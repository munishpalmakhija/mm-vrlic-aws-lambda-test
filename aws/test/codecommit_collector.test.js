/*
Copyright 2018 VMware, Inc.
SPDX-License-Identifier: MIT
*/
const { lintTestEnv } = require('./helper.test');
const { codeCommitSampleData } = require('./sample_files/codecommit_testdata');

const {
  CodeCommitHttpCollector,
} = require('../src/index');


describe('CodeCommitHttpCollector', () => {
  const collector = new CodeCommitHttpCollector(lintTestEnv);

  describe('processLogsJson', () => {
    it('should process logs JSON', () => {
      const logsJson = codeCommitSampleData();
      collector.processLogsJson(logsJson);
      expect(logsJson).toMatchSnapshot();
    });
  });
});

