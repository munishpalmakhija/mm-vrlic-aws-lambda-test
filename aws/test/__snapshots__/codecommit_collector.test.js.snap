// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeCommitHttpCollector processLogsJson should process logs JSON 1`] = `
Object {
  "Records": Array [
    Object {
      "awsRegion": "us-east-2",
      "codecommit": Object {
        "references": Array [
          Object {
            "commit": "694e945e6f475db634090f35386f72b9d5279ae4",
            "created": true,
            "ref": "refs/heads/Test",
          },
        ],
      },
      "customData": null,
      "eventId": "15382760-edef-487d-97d8-de1b976fca4b",
      "eventName": "ReferenceChanges",
      "eventPartNumber": 1,
      "eventSource": "aws:codecommit",
      "eventSourceARN": "arn:aws:codecommit:us-east-2:827066587692:codepipe",
      "eventTime": "2020-09-22T12:32:45.075+0000",
      "eventTotalParts": 1,
      "eventTriggerConfigId": "897e3f3f-aaec-442a-851e-e668502c46aa",
      "eventTriggerName": "codeCommitTrigger",
      "eventVersion": "1.0",
      "ingest_timestamp": 1538769915817,
      "log_type": "aws_codecommit",
      "userIdentityARN": "arn:aws:iam::827066587692:user/sample_iam_user",
    },
  ],
}
`;
